lib/puppet/type/consul_prepared_query.rb

```diff
@@ -108,4 +108,18 @@
       raise ArgumentError, "Number of API tries must be a number" if not value.is_a?(Integer)
     end
   end
+
+  newparam(:template_regexp) do
+    desc 'regexp for template'
+    validate do |value|
```

Comment:
Yeah, for sure. I'm somewhat new to ruby types so I'll read the docs a bit more and tweak this.